extensions/amp-story/1.0/amp-story.js

```diff
@@ -2216,6 +2240,27 @@ export class AmpStory extends AMP.BaseElement {
     if (!this.sidebar_) {
       return;
     }
+    if (isExperimentOn(this.win,'amp-story-branching')) {
+      this.sidebar_.addEventListener('click', e => {
+
+        if (e.target.tagName === 'A') {
+          const url = e.target.getAttribute('href');
+          if (url.indexOf('#page=') >= 0) {
+            // Do not let the browser scroll
+            e.preventDefault();
+
+            // Handle for absolute URLs, in addition to fragments only.
+            this.win.location.hash = url.slice(url.search('\#(.*)'));
+
+            const actions = Services.actionServiceForDoc(this.element);
+            actions.execute(dev().assertElement(this.sidebar_),
+                'close', /* args */ null, /* source */ null, /* caller */ null,
+                /* event */ null, ActionTrust.HIGH);
+            this.closeOpacityMask_();
+          }
+        }
+      });
+    }
```

Comment:
A few things about this block:

- If the link is meant to navigate to another story, this will fail. ie: sidebar in `story.com/1` links to `story.com/2#page=foo` would not navigate to `story.com/2`
- This will fail if there is any element within the `<a>` tag. ie: `<a href="#page=foo">Foo</a>` will work, but `<a href="#page=foo"><span>Foo</span></a>` won't
- Does this piece of code have the same history state issue described above?

Could we do all that from the `hashchange` handler?

Reply:
We took this offline a little, but done! Utilized hashchange handler for fragments, but not for goToPage. I also added a check for the pathname, to handler for href values that look like `#page=somePage` or `story.com/2#page=somePage` if the story was on `story.com/2`